Tests/Services/ParkingSlotServiceTests.cs

```diff
@@ -266,58 +266,38 @@ public void GivenZoneIdStartTimeAndDuration_WhenGetFreeByZoneIdAndTimePeriodCall
         #endregion
 
         #region IsSlotFree
-        [Fact]
-        public void GivenSlotStartTimeAndDuration_WhenIsSlotFreeCalled_ThenFalseReturned()
+        public static IEnumerable<object[]> TestData()
         {
-            //Arrange
-            var testStartTime = new DateTime(2024, 1, 27, 18, 00, 00);
-            var testDuration = 4;
             var slot = new ParkingSlot()
             {
                 Id = new Guid("d4b5425b-a731-4f5d-a61c-f9441fc388d5"),
                 Reservations = new List<Reservation>()
                 {
                     new()
                     {
-                        Id = _testReservationId,
                         StartTime = new DateTime(2024, 1, 27, 16, 00, 00),
-                        Duration = 5
+                        Duration = 2
                     }
                 }
```

Comment:
Add one more Reservation with StartTime 21:00 and duration 2. I expect that in this case your tests on lines 284 and 285 would fail, though they mustn't.